src/config.rs

```diff
@@ -362,6 +364,16 @@ impl Cfg {
         Toolchain::from(self, name)
     }
 
+    pub fn get_toolchains_from_glob(
+        &self,
+        pattern: Pattern,
+    ) -> Result<impl Iterator<Item = Toolchain<'_>>> {
+        Ok(self
+            .list_toolchains_iter()?
+            .filter(move |toolchain| pattern.matches(toolchain))
+            .map(move |toolchain| Toolchain::from(self, &toolchain).unwrap()))
```

Comment:
Why does this closure need `move` ?

Reply:
I think because otherwise it reborrows `self`? I don't know enough about how reborrows work, but this is the error:

```
error[E0373]: closure may outlive the current function, but it borrows `self`, which is owned by the current function
   --> src/config.rs:374:18
    |
374 |             .map(|toolchain| Toolchain::from(self, &toolchain).unwrap()))
    |                  ^^^^^^^^^^^                 ---- `self` is borrowed here
    |                  |
    |                  may outlive borrowed value `self`
    |
note: closure is returned here
   --> src/config.rs:370:10
    |
370 |     ) -> Result<impl Iterator<Item = Toolchain<'_>>> {
    |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
help: to force the closure to take ownership of `self` (and any other referenced variables), use the `move` keyword
    |
374 |             .map(move |toolchain| Toolchain::from(self, &toolchain).unwrap()))
    |                  ^^^^^^^^^^^^^^^^
```

Reply:
Aah because the iterator gets returned, fair enough.